### ISSUE TITLE
Loads same entries and doesn't load the final entry available

### ISSUE THREAD

**Issue by CreateSean** [putyourlightson/craft-sprig#12]
I'm trying out the load-more recipe for sprig and for the most part it is working.

On page load the first 3 items are there (would like to have zero loaded but I can work with this). Then when I click the load more button 3 more entries load, but they are the same as the initial 3. Then when I click load more nothing happens despite there being 7 test entries in the channel.

See gif below and my code.

**Main Template**
```
  <section class="py-68">
    {# Creates a component from the template path #}
  {{ sprig('_components/load-more', {
      _offset: 0,
      _limit: 3,
  }, {class: 'flex flex-row flex-wrap w-full'}) }}

  {# Loads the required script from a CDN #}
  {{ sprig.script }}
  </section>
```
**Load More Template**
```
{% set entries = craft.entries.section('blog').offset(_offset).limit(_limit).all() %}
{# show the entries #}
{% for entry in entries %}  
      {% include 'blog/_listing-entry' %}
{% endfor %}


{# load the entries here #}
{% if entries %}
  <div class="container flex flex-row flex-wrap w-full"> 
      <div id="replace" class="flex flex-row flex-wrap w-full">
          <input type="hidden" name="offset" value="{{ _offset + _limit }}">
          <p class="w-full text-center">
            <button sprig s-target="#replace" s-swap="outerHTML" class="mx-auto btn btn-white">{{ "Load More"|t }}</button>
          </p><!-- /.text-center -->
      </div>
  </div><!-- /.container -->
{% endif %}
```

And the animated gif

![sprig-load-more-2](https://user-images.githubusercontent.com/6899490/88954905-19888d80-d269-11ea-89b0-1ac35074940a.gif)

**Comment by bencroker** [comment]
You shouldn't be using an underscore in the names of variables unless you want them to be "protected" from manipulation. Removing the underscore should make it work.

If you want to show no results on initial page load then you can listen for `sprig.request` as per the example at https://putyourlightson.com/sprig-cookbook#dynamic-content.

**Comment by CreateSean** [comment]
@bencroker 

removed underscore and now loads more entries but won't load the final entry in the channel. I can see 6 of 7 entries total.

I then changed to this code modelled after the dynamic-content link you shared (code from docs did not work for me) -  changed it when it also only showed the first three entries and the load more button didn't work. Still doesn't work.

<div class="block w-full">
  {% if sprig.request %}
    
    {% set entries = craft.entries.orderBy('RAND()').limit(limit).all() %}

    {% for entry in entries %}
        <h6>{{ entry.title }}</h6><br>
    {% endfor %}
{% endif %}

  <div sprig s-trigger="load" class="w-full text-center"><button sprig s-target="#replace" s-swap="outerHTML" class="mx-auto btn btn-white">{{ "Load More"|t }}</button></div>
</div>


Finally tried this and again the first 3 entries display on load but the load more button doesn't work

```
<div class="container flex flex-row flex-wrap w-full"> 
    <div id="replace" class="flex flex-row flex-wrap w-full">

      {% if sprig.request %}
        
        {% set entries = craft.entries.section('blog').offset(offset).limit(limit).all() %}

        {% for entry in entries %}
            {% include 'blog/_listing-entry' %}
        {% endfor %}
         
      {% endif %}

      <div sprig s-trigger="load" class="container w-full">
        <p class="w-full text-center">
          <button sprig s-target="#replace" s-swap="outerHTML" class="mx-auto btn btn-white">{{ "Load More"|t }}</button>
        </p><!-- /.text-center -->
      </div>

    </div>
</div><!-- /.container -->
```

**Comment by bencroker** [comment]
Below is a combination of the [load more](https://putyourlightson.com/sprig-cookbook#load-more) and [dynamic-content](https://putyourlightson.com/sprig-cookbook#dynamic-content) examples that should get you what you're after.

```twig
{% set offset = offset ?? 0 %}
{% set limit = limit ?? 3 %}

{% if not sprig.request %}

  <div sprig s-trigger="load"></div>

{% else %}

  {% set entries = craft.entries.offset(offset).limit(limit).all() %}

  {% for entry in entries %}
    <h5>{{ entry.title }}</h5>
  {% endfor %}

  {% if entries %}
    <div s-target="this" s-swap="outerHTML">
      <input type="hidden" name="offset" value="{{ offset + limit }}">
      <button sprig>Load another</button>
    </div>
  {% endif %}

{% endif %}
```

**Comment by CreateSean** [comment]
@bencroker I tried that and I get a list of 3 items but the load more button does nothing.

**Comment by CreateSean** [comment]
@bencroker I've determined that the error was due to laravel mix proxying my local site domain to localhost:3000 instead of the actual local domain. Once I visited at the correct domain the load more worked correctly as expected.

Determined this after checking the console and seeing:

`Access to XMLHttpRequest at 'https://client-site.ddev.site/index.php?p=actions/sprig/components/render&offset=6&sprig%3Atemplate=f7415b104db644e873377b83734c64cfe3668fe2db36c09f76fc759bfea584d2_components%2Fload-more&sprig%3Avariables%5Blimit%5D=17dbc94ba93b1142e4c952445362c7452d811124dc4bf6222eb6477aca0100113&sprig%3Avariables%5Boffset%5D=9e85083ea544f3c4d0fa0a6d53aada8f32caf7a1c8c8ff56243d0953848ce9d10' from origin 'https://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

At this point I feel that all is good.
